### PR TITLE
Add support for logging w/ per-request metadata.

### PR DESCRIPTION
**Pull request by tseaver**
- `insert_id`
- `severity`
- `http_request`

See: #1566.

### CODE REVIEW

File: gcloud/logging/entries.py [hunk 1]
```diff
@@ -40,14 +40,23 @@ class _BaseEntry(object):
 
     :type labels: dict or :class:`NoneType`
     :param labels: (optional) mapping of labels for the entry
+
+    :type severity: string or :class:`NoneType`
+    :param severity: (optional) severity of event being logged.
+
+    :type http_request: dict or :class:`NoneType`
+    :param http_request: (optional) info about HTTP request associated with
```
Comment (dhermes) [inline gcloud/logging/entries.py]: Why is this needed? If it's a `dict` wouldn't there be a better name the `http_request`?

Comment (dhermes) [inline gcloud/logging/entries.py]: Ahhh I didn't realize `httpRequest` was a thing

File: gcloud/logging/logger.py [hunk 2]
```diff
@@ -291,10 +349,22 @@ def log_text(self, text, labels=None):
...
+        :type severity: string or :class:`NoneType`
+        :param severity: (optional) severity of event being logged.
+
+        :type http_request: dict or :class:`NoneType`
+        :param http_request: (optional) info about HTTP request associated with
+                             the entry.
         """
-        self.entries.append(('text', text, labels))
+        self.entries.append(
+            ('text', text, labels, insert_id, severity, http_request))
```
Comment (dhermes) [inline gcloud/logging/logger.py]: This is starting to become cumbersome. WDYT? (`collections.namedtuple`?)

Comment (tseaver) [inline gcloud/logging/logger.py]: I thought about it, but it wouldn't help that much:  it would actually make *this* site *more* verbose.

**Comment by tseaver** [comment]
@dhermes any remaining issues?

**Comment by dhermes** [comment]
LGTM
